core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/dom/parser/AbstractGNUSourceCodeParser.java

```diff
@@ -1109,7 +1114,7 @@ private IASTExpression buildExpression(IASTExpression left, BinaryOperator opera
 			break;
 		case IToken.tAND:
 			op = IASTBinaryExpression.op_logicalAnd;
-			unaryOp = IASTUnaryExpression.op_labelReference;
+			unaryOp = IGNUASTUnaryExpression.op_labelReference;
```

Comment:
_This comment goes with the comment about `IGNUASTUnaryExpression` not being deprecated anymore._

This leads to a strange case as `unaryOp` is passed to (eventually) the constructor of one of the concrete classes of `IASTUnaryExpression` (see call hierarchy of `INodeFactory.newUnaryExpression(int, IASTExpression)`

I suppose to complete the split there should be a new subclass of org.eclipse.cdt.internal.core.dom.parser.NodeFactory which is GNU vs non-GNU specific.